feat: nebula decoder update

citruscosmos opened the pull request:
## Description
The primary goal is to replace the existing drs_seyond.launch.xml with the new drs_nebula_decoder.launch.xml in the offline launch configuration (drs_offline.launch.xml). This change supports decoding LiDAR data from sensors like Falcon and Robinw using the common Nebula driver framework.



## Key Changes

- New Launch File: Added src/drs_launch/launch/component/drs_nebula_decoder.launch.xml to configure and launch the nebula_decoder_node_exe.

- Offline Launch Update: Modified drs_offline.launch.xml to use the new drs_nebula_decoder.launch.xml for all LiDAR units (front, right, rear, left).

- Dependencies (package.xml): Added nebula_decoder as an <exec_depend> in src/drs_launch/package.xml to ensure the new dependency is met.

- Repositories (drs.repos):
  - Added the bag_converter repository (tier4/bag_converter.git).
  - (merge from the latest develop/r36.4.0) Updated several repository URLs (e.g., accelerated_image_processor) from git@ to https.
  - (merge from the latest develop/r36.4.0) Pinned all dependencies to specific commit hashes to ensure reproducible builds.

## Verification
- Generated the runtime Docker image and performed local build to ensure the build is successful.
- Conducted conversion tests using data that was problematic with the old offline launch configuration (which produced abnormal LiDAR point cloud timestamps, such as 0).
  - Confirmed: With the new nebula_decoder configuration, the same data is converted correctly, and the resulting point cloud timestamps are valid (no abnormal values).

0x126 committed:
feat: add drs_nebula_decoder launch file and update drs_offline launch to include it

0x126 committed:
fix: update package.xml

citruscosmos committed:
fix: update drs.repos to use tier4 bag_converter and reflect the recent update

otenim commented on drs.repos:
```diff
@@ -2,36 +2,40 @@ repositories:
+    version: 058feb116fecdaccfbdc47571ba98400be3dfe7d  # main
+  bag_converter:
+    type: git
+    url: git@github.com:tier4/bag_converter.git
```
publicなレポジトリにはgitから始まるurlは設定しないでください。
sshキーを設定していないとcloneできなくなります。
private -> git~
public -> https~
としてください。

citruscosmos replied:
承知しました。
また、修正して頂き、ありがとうございます。
次の更新時気をつけます。

otenim replied:
もう一点、bag_converterの依存パッケージであるseyond_sdkが入っておらずビルドに失敗するのでdrs.reposに入れておきました。

otenim replied:
vcstoolの微妙なところは、他のパッケージの中にある.reposファイルを再帰的に見て勝手にクローンしてくれるわけじゃないので、トップレベルのパッケージの.reposファイルで全て指定する必要があります。

citruscosmos replied:
申し訳ありません、Seyondドライバの対応で同様にseyond_sdkを入れる対応をしていたのですが、こちらの修正ちゃんとチェック出来ておりませんでした。

otenim committed:
refactor: use https prefix for public repos

otenim committed:
Merge branch 'develop/r36.4.0' into feat/nebula_decoder

citruscosmos commented:
間違えて平林さんをレビュワーに設定してしまっていたため修正しました。

otenim committed:
chore: update bag_converter version to 2413e30

otenim committed:
refactor: add seyond_sdk (dependency of bag_converter)

otenim committed:
docs: add comprehensive system overview to README

otenim committed:
edit: fixed typo

otenim committed:
build: add ros, ros2 to the dictionary

otenim committed:
ci: remove redundant cspell-json-url from workflow

The URL is already imported in .cspell.json, so specifying it again
in the workflow is redundant. This simplifies configuration management.

otenim committed:
Revert "ci: remove redundant cspell-json-url from workflow"

This reverts commit 4bcf7a3f6027acde9692ee73d7a4b839f252ad44.

otenim committed:
chore: add project-specific words to cspell dictionary

otenim committed:
docs: fix ROS 2 naming style in README

otenim commented:
@copilot Revise summary of this PR.

otenim committed:
refactor: formatting

otenim commented:
@citruscosmos 一応READMEに書いてある通りにビルドして確かめて、他もOKそうだったらマージしちゃってください（READMEも書き直しました）

citruscosmos commented:
@otenim READMEに書いてある通りのビルドは通りました。ローカルでビルドした環境、runtime docker環境共にoffilne decoderの動作も確認しました。
